dags/analyzer_helper/tests/unit/test_unit_discord_transform_raw_data.py

```diff
@@ -0,0 +1,129 @@
+import unittest
+from datetime import datetime
+
+from analyzer_helper.discord.discord_transform_raw_data import DiscordTransformRawData
+
+
+class DiscordTransformRawDataUnitTest(unittest.TestCase):
+
+    def test_create_interaction_valid_data(self):
+        transformer = DiscordTransformRawData()
+        interaction = transformer.create_interaction("reply", ["user1234"], "emitter")
+        self.assertEqual(
+            interaction,
+            {
+                "name": "reply",
+                "users_engaged_id": ["user1234"],
+                "type": "emitter",
+            },
+        )
+
+    def test_create_interaction_missing_arguments(self):
+        transformer = DiscordTransformRawData()
+        with self.assertRaises(TypeError):
+            transformer.create_interaction("reply")
+
+    def test_create_receiver_interaction_valid_data(self):
+        transformer = DiscordTransformRawData()
+        data = {
+            "createdDate": {"$date": "2024-06-11T00:00:00Z"},
+            "messageId": "12345",
+            "threadId": "abc",
+            "channelId": "xyz",
+            "isGeneratedByWebhook": False,
+            "botActivity": False,
+        }
+        interaction = transformer.create_receiver_interaction(
+            data, "reply", "user456", "user789"
+        )
+        self.assertEqual(
+            interaction,
+            {
+                "author_id": "user456",
+                "date": "2024-06-11T00:00:00Z",
+                "source_id": "12345",
+                "metadata": {
+                    "thread_id": "abc",
+                    "channel_id": "xyz",
+                    "bot_activity": False,
+                },
+                "actions": [],
+                "interactions": [
+                    {
+                        "name": "reply",
+                        "users_engaged_id": ["user789"],
+                        "type": "receiver",
+                    }
+                ],
+            },
+        )
+
+    def test_create_emitter_interaction_valid_data(self):
+        transformer = DiscordTransformRawData()
+        data = {
+            "messageId": "56789",
+            "threadId": "def",
+            "channelId": "ghi",
+            "isGeneratedByWebhook": True,
+            "botActivity": True,
+        }
```

Comment:
The rawinfos data should have all fields within it. ref: https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=An%20example%20for%20rawinfos%20document